### PR TITLE
Extract components from page layout

### PR DESCRIPTION
**Pull request by michael-coker**
fixes https://github.com/patternfly/patternfly-next/issues/637

@andresgalante I left the JS that toggles the navigation in the demo. It's easy to remove if we want to. For the sake of consistency and for future demos, I don't think we should write javascript in our workspace, and just present the different states instead. To see the interaction/animations, maybe link to a demo of the react components in a sample application since that's officially supported. What do you think?

### CODE REVIEW

File: src/patternfly/demos/PageLayout/examples/page-layout-default-nav-example.hbs [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
     {{/page-header-brand}}
-    {{#> page-template-header-elements}}
-    {{/page-template-header-elements}}
+    {{#> page-template-header-tools-elements}}
+    {{/page-template-header-tools-elements}}
   {{/page-header}}
-
   {{!-- Nav --}}
   {{#> page-sidebar}}  
     {{#> nav nav--vertical="true" nav--attribute='id="primary-nav" aria-label="Primary Nav Default Example"'}}
```
Comment (mattnolting) [inline src/patternfly/demos/PageLayout/examples/page-layout-default-nav-example.hbs]: nav--vertical="true" is no longer needed

File: src/patternfly/layouts/Page/styles.scss [hunk 2]
```diff
@@ -200,10 +191,15 @@ $nav: "pf-c-nav";
   margin-left: auto; // to push it to the right
 
   .pf-l-toolbar {
-    // include margin-top for avatar to position itself vertically
-    margin-top: var(--pf-l-page__header-tools--toolbar--MarginTop);
+    margin-top: var(--pf-l-page__header-tools--toolbar--MarginTop); // include margin-top for avatar to position itself vertically
```
Comment (mattnolting) [inline src/patternfly/layouts/Page/styles.scss]: Nice, I like this!

File: src/patternfly/components/Avatar/styles.scss [hunk 3]
```diff
@@ -1,19 +1,7 @@
...
 .pf-c-avatar {
-  /* stylelint-disable */
-
-  // Header avatar
-  --pf-l-page__header-avatar--Width:                        pf-size-prem(48px);
-  --pf-l-page__header-avatar--Height:                       pf-size-prem(48px);
-  --pf-l-page__header-avatar--MarginLeft:                   var(--pf-global--spacer--md);
-
-  /* stylelint-enable */
+  --pf-c-avatar--BorderRadius: 50%;
```
Comment (matthewcarleton) [inline src/patternfly/components/Avatar/styles.scss]: can't we use `--pf-global--BorderRadius--lg` here?

**Comment by mattnolting** [comment]
@michael-coker @andresgalante I agree that best practice is to present different states. With more complex components, it's just a lot more work to do so when including very basic JS makes it much easier to understand. It doesn't seem to be a big issue while we're still in Alpha, but will be in Beta+. Insights will be using all interactive components, so that would be a good opportunity to link to a live demo when they become available.

File: src/patternfly/demos/PageLayout/page-template-header-brand-elements.hbs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{{#if page-template-header-brand--toggle}}
+  {{#> button btnClass="pf-m-plain"
```
Comment (matthewcarleton) [inline src/patternfly/demos/PageLayout/page-template-header-brand-elements.hbs]: we should probably update btnClass to be button--modifier. I'll create an issue for that though.

File: src/patternfly/layouts/Page/styles.scss [hunk 5]
```diff
@@ -4,97 +4,86 @@
...
-  --pf-l-page__header-brand--PaddingBottom--sm:             var(--pf-global--spacer--md);
-  --pf-l-page__header-brand--BackgroundColor--md:           var(--pf-global--BackgroundColor--dark-transparent-100);
-  // Hardcode header brand width based on design spec, must match --pf-c-nav--Width
-  --pf-l-page__header-brand--FlexBasis--lg:                 pf-size-prem(290px);
+  --pf-l-page__header-brand--PaddingTop: var(--pf-global--spacer--md);
+  --pf-l-page__header-brand--PaddingBottom: var(--pf-global--spacer--md);
+  --pf-l-page__header-brand--PaddingLeft: var(--pf-global--spacer--xl);
+  --pf-l-page__header-brand--PaddingRight: var(--pf-global--spacer--xl);
+  --pf-l-page__header-brand--PaddingBottom--sm: var(--pf-global--spacer--md);
+  --pf-l-page__header-brand--BackgroundColor--md: var(--pf-global--BackgroundColor--dark-transparent-100); // Hardcode header brand width based on design spec, must match --pf-c-nav--Width
```
Comment (matthewcarleton) [inline src/patternfly/layouts/Page/styles.scss]: should this comment be on the next line?

File: src/patternfly/layouts/Page/styles.scss [hunk 6]
```diff
@@ -4,97 +4,86 @@
...
+  --pf-l-page__header-nav--MarginLeft--md: var(--pf-global--spacer--xl);
+  --pf-l-page__header-nav--MarginRight--md: var(--pf-global--spacer--xl);
+  --pf-l-page__header-nav--BackgroundColor--sm: var(--pf-global--BackgroundColor--dark-transparent-100);
+
+  // Header tools
+  --pf-l-page__header-tools--MarginRight: var(--pf-global--spacer--xl);
+  --pf-l-page__header-tools--toolbar--MarginTop: var(--pf-global--spacer--md);
+  --pf-l-page__header-tools--toolbar--MarginBottom: var(--pf-global--spacer--md);
+  --pf-l-page__header-tools--avatar--MarginLeft: var(--pf-global--spacer--md);
+  --pf-l-page__header-tools--avatar--Width: pf-size-prem(48px);
```
Comment (andresgalante) [inline src/patternfly/layouts/Page/styles.scss]: Why is this hardcoded? can we add a comment to explain?

File: src/patternfly/layouts/Page/styles.scss [hunk 7]
```diff
@@ -130,29 +119,31 @@ $nav: "pf-c-nav";
...
-    align-items: flex-end;
     align-self: stretch;
     padding-right: var(--pf-l-page__header-brand--PaddingRight); // set padding right here to allow mobile view to accomodate tools
     background-color: var(--pf-l-page__header-brand--BackgroundColor--md);
   }
 
-  .pf-m-patternfly-logo {
-    height: var(--pf-l-page__header-brand-logo--Height);
+  // Sidebar Toggle
+  .pf-c-button {
```
Comment (andresgalante) [inline src/patternfly/layouts/Page/styles.scss]: Also a comment here to explain why we are changing the pf-c-button within the sidebar.
Just a thoight: we don't want anyone to add a normal btn there, but is this a dangerous move? what is a user wants to add a btn to this section and make it look like a button.

File: src/patternfly/layouts/Page/styles.scss (hunk 6 above)
Comment (michael-coker) [reply src/patternfly/layouts/Page/styles.scss]: updated

File: src/patternfly/layouts/Page/styles.scss [hunk 8]
```diff
@@ -130,29 +119,31 @@ $nav: "pf-c-nav";
     display: flex;
```
[the remark this answers is not in the document]
Comment (michael-coker) [reply src/patternfly/layouts/Page/styles.scss]: I don't see a comment around here?

File: src/patternfly/layouts/Page/styles.scss (hunk 7 above)
Comment (michael-coker) [reply src/patternfly/layouts/Page/styles.scss]: Added a container in the layout for the toggle.

File: src/patternfly/layouts/Page/styles.scss (hunk 5 above)
Comment (michael-coker) [reply src/patternfly/layouts/Page/styles.scss]: yes! good catch.

File: src/patternfly/demos/PageLayout/examples/page-layout-default-nav-example.hbs (hunk 1 above)
Comment (michael-coker) [reply src/patternfly/demos/PageLayout/examples/page-layout-default-nav-example.hbs]: removed

File: src/patternfly/components/Avatar/styles.scss (hunk 3 above)
Comment (michael-coker) [reply src/patternfly/components/Avatar/styles.scss]: doh! yes 👍

**Pull request merged**